tests/test_callbacks.py

```diff
@@ -36,9 +44,76 @@ def test_callbacks():
     check_estimators_pickle("KerasClassifier", estimator)
     check_estimators_partial_fit_n_features("KerasClassifier", estimator)
     # Check that callback was called
-    estimator.fit([[0]], [1])  # quick fit
-    assert estimator.callbacks[0].called != SentinalCallback.called
+    estimator.fit([[0]], [1])
+    assert estimator.callbacks[0].train_counter == 1
     serialized_estimator = pickle.dumps(estimator)
     deserialized_estimator = pickle.loads(serialized_estimator)
-    assert deserialized_estimator.callbacks[0].called == estimator.callbacks[0].called
-    estimator.fit([[0]], [1])  # quick fit
+    assert deserialized_estimator.callbacks[0].train_counter == 1
+    deserialized_estimator.fit([[0]], [1])
+    assert deserialized_estimator.callbacks[0].train_counter == 2
+
+
+class Schedule:
+    """Simple exponential decay lr scheduler.
+    Modeled after https://keras.io/api/callbacks/learning_rate_scheduler/
+    Used to test nested object construction for callbacks.
+    """
+
+    def __init__(self, coef: float = 0.1) -> None:
+        self.coef = coef
+
+    def __call__(self, epoch: int, lr: float):
+        if epoch > 10:
+            lr = lr * math.exp(-self.coef)
+        return lr
+
+
+@pytest.mark.parametrize("prefix", ("", "fit__"), ids=["no prefix", "fit__ prefix"])
+@pytest.mark.parametrize(
+    "callback_kwargs",
+    [
+        dict(
+            callbacks=[keras.callbacks.LearningRateScheduler],
+            callbacks__0__0=Schedule,  # LRS does not acccept kwargs, only args, hence the 0__0 syntax
+            callbacks__0__0__coef=0.2,  # translates to kwarg "coef" to the first arg of the first element of the callbacks kwarg to fit
+        ),
+        dict(
+            callbacks={"lrs": keras.callbacks.LearningRateScheduler},
+            callbacks__lrs__0=Schedule,  # here the __0 is again because LRS does not accept kwargs
+            callbacks__lrs__0__coef=0.2,
+        ),
```

Comment:
@stsievert just want to check before I merge: does this look good to you? Thanks